app/UI/relay-recommend-list.tsx

```diff
@@ -9,8 +9,40 @@ import {
     HoverButtonBackgroudColor,
     PrimaryTextColor,
 } from "./style/colors.ts";
+import { emitFunc } from "../event-bus.ts";
+import { RelayConfigChange } from "./setting.tsx";
+import { RelayConfig } from "./relay-config.ts";
+
+type RelayRecommendListProps = {
+    relayConfig: RelayConfig;
+    emit: emitFunc<RelayConfigChange>;
+};
+
+type RelayRecommendListState = {
+    relays: string[];
+};
+
+export class RelayRecommendList extends Component<RelayRecommendListProps, RelayRecommendListState> {
+    state: Readonly<RelayRecommendListState> = {
+        relays: this.computeRecommendedRelays(),
+    };
+
+    computeRecommendedRelays() {
+        // remove the relays that are already in the relayConfig
+        return recommendedRelays.filter((r) => !this.props.relayConfig.getRelayURLs().has(r));
+    }
+
+    handleAddRelay = (relayUrl: string) => {
+        // There is no need to get the relay status here
+        this.props.relayConfig.add(relayUrl);
+        this.props.emit({
+            type: "RelayConfigChange",
+            kind: "add",
+            url: relayUrl,
+        });
+        this.setState({ relays: this.computeRecommendedRelays() });
+    };
```

Comment:
Make this function async and handle all the error cases. Error handling is very important. If you choose to not handle intentionally, always put a comment inline.